src/main/java/edu/ucla/library/libcal/handlers/ProxyHandler.java

```diff
@@ -69,19 +79,25 @@ public ProxyHandler(final Vertx aVertx) {
     public void handle(final RoutingContext aContext) {
         final HttpServerResponse response = aContext.response();
         final String path = aContext.request().path();
-
-        final String receivedQuery =
-                path.concat(aContext.request().query() != null ? QUESTION_MARK.concat(aContext.request().query()) : "");
-        myTokenProxy.getBearerToken().compose(token -> {
-            return myApiProxy.getLibCalOutput(token, SLASH.concat(receivedQuery)).onSuccess(apiOutput -> {
-                response.setStatusCode(apiOutput.getInteger(JsonKeys.STATUS_CODE));
-                response.setStatusMessage(apiOutput.getString(JsonKeys.STATUS_MESSAGE));
-                response.putHeader(HttpHeaders.CONTENT_TYPE, APPLICATION_JSON.toString());
-                response.end(apiOutput.getString(JsonKeys.BODY));
+        final String originalClientIP = aContext.request().headers().get(Constants.X_FORWARDED_FOR).split(COMMA)[0];
```

Comment:
```suggestion
        final String originalClientIP = aContext.request().remoteAddress().hostAddress();
```

If you use this instead, you get the X-Forwarded-For parsing reading for free (since you've got the server router configured with `router.allowForward(AllowForwardHeaders.X_FORWARD);`) while also handling the case where there's no reverse proxy (e.g. if a dev wants to simply run a standalone container on their local).

Reply:
Nice. Making note for myself, too.

Reply:
Switched to `remoteAddress().hostAddress()`.